shared/extension/libraries.cmake

```diff
@@ -3,3 +3,7 @@ find_package(yaml-cpp REQUIRED)
 target_link_libraries(nuclear_extension PUBLIC yaml-cpp)
 
 target_compile_features(nuclear_extension PUBLIC cxx_std_17)
+
+# Set warnings
+include(CompilerWarnings)
+set_target_warnings(nuclear_extension)
```

Comment:
I have decided you can move these into NUClear proper so long as you add a cache variable thing so that these warnings are configurable from outside of NUClear